### PR TITLE
"Prototype an assistant" page with integrated editors, train and download buttons.

### PR DESCRIPTION
**Pull request by alwx**
Requires Rasabaster changes: https://github.com/RasaHQ/rasabaster/pull/73

**Proposed changes**:
- adds "Prototype and assistant" page with ACE editors, train and download buttons

**Status (please check what you already did)**:
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -201,3 +200,7 @@ git = "https://github.com/RasaHQ/sphinxcontrib-versioning.git"
 
 [tool.poetry.dev-dependencies.sphinx_rtd_theme]
 git = "https://github.com/RasaHQ/sphinx_rtd_theme.git"
+
+[tool.poetry.dev-dependencies.rasabaster]
+git = "https://github.com/RasaHQ/rasabaster.git"
+branch = "prototyper"
```
Comment (alwx) [inline pyproject.toml:206]: this one needs to be removed once rasabaster change is merged

File: pyproject.toml [hunk 2]
```diff
@@ -156,7 +156,7 @@ sphinxcontrib-websupport = "^1.1.0"
 sphinxcontrib-trio = "==1.1.2"
 sphinx-tabs = "~1.1.13"
 sphinx-autodoc-typehints = "==1.10.3"
-rasabaster = "^0.7.23"
+rasabaster = "^0.7.28"
```
Comment (m-vdb) [inline pyproject.toml:159]: CI doesn't seem to be able to install this version

File: docs/conf.py [hunk 3]
```diff
@@ -36,6 +37,10 @@
     "rasabaster.editlink",
     "rasabaster.runnable",
     "rasabaster.conversations",
+    "rasabaster.ace",
+    "rasabaster.train_button",
+    "rasabaster.download_button",
+    "rasabaster.chat_block",
 ]
```
Comment (erohmensing) [inline docs/conf.py:44]: @m-vdb will you add these to the extensions list for the docs switch?

File: docs/index.rst [hunk 4]
```diff
@@ -25,6 +25,7 @@ Understand messages, hold conversations, and connect to messaging channels and A
    user-guide/installation
    user-guide/rasa-tutorial
    user-guide/building-assistants
+   user-guide/prototype-an-assistant
```
Comment (erohmensing) [inline docs/index.rst:28]: @abhilasharoy am I correct in saying this was supposed to replace the rasa tutorial?

File: docs/user-guide/prototype-an-assistant.rst [hunk 5]
```diff
@@ -0,0 +1,198 @@
+Prototype an Assistant
+======================
+
+.. edit-link::
+
+Get started with Rasa Open source and learn how to create an assistant from scratch!
```
Comment (erohmensing) [inline docs/user-guide/prototype-an-assistant.rst:6]:
```suggestion
Get started with Rasa Open Source and learn how to create an assistant from scratch!
```

File: docs/user-guide/prototype-an-assistant.rst [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+
+This page explains the basics of building an assistant with Rasa and shows the structure of a Rasa project.
+You can test it out right here without installing anything. You can also install Rasa and follow along in your command line.
```
Comment (erohmensing) [inline docs/user-guide/prototype-an-assistant.rst:9]:
```suggestion
This page explains the basics of building an assistant with Rasa Open Source and shows the structure of a Rasa project.
You can test it out right here without installing anything. You can also :ref:`install Rasa Open Source <installation>` and follow along in your command line.
```

File: docs/user-guide/prototype-an-assistant.rst [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+
+
+1. Define a basic user goal
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To prototype an assistant, start with a single `user goal <glossary>`_
```
Comment (erohmensing) [inline docs/user-guide/prototype-an-assistant.rst:15]:
```suggestion
To prototype an assistant, start with a single :ref:`user goal <glossary>`
```

File: docs/user-guide/prototype-an-assistant.rst [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+that your assistant will handle. You can create a prototype using the user goal we have chosen
+"subscribing to a newsletter", or you can come up with your own. If you choose your own user goal,
```
Comment (erohmensing) [inline docs/user-guide/prototype-an-assistant.rst:17]:
```suggestion
that your assistant will handle. You can create a prototype with the "subscribing to a newsletter" user goal we have chosen,
or you can come up with your own. If you choose your own user goal,
```

File: docs/user-guide/prototype-an-assistant.rst [hunk 9]
```diff
@@ -0,0 +1,198 @@
...
+you can create a prototype that handles your goal by following the instructions to change each
+section throughout the tutorial.
+
+
+2. Create some NLU data
+~~~~~~~~~~~~~~~~~~~~~~~
+
+`NLU <glossary>`_ data provides examples of what users might say to your assistant and what they mean by it.
+`Intent <glossary>`_  refers to what the user means with a specific message.
```
Comment (erohmensing) [inline docs/user-guide/prototype-an-assistant.rst:26]:
```suggestion
:ref:`NLU <glossary>` data provides examples of what users might say to your assistant and what they mean by it.
An :ref:`intent <glossary>` refers to what the user means with a specific message.
```

File: docs/user-guide/prototype-an-assistant.rst [hunk 10]
```diff
@@ -0,0 +1,198 @@
...
+.. code-editor::
+    :language: yaml
+    :id: forms
+    :height: 150
+    :tracking_endpoint: https://trainer-service.prototyping.rasa.com/startPrototyping
+
+    forms:
+    - newsletter_form:
+       email:
+       - type: from_text
```
Comment (erohmensing) [inline docs/user-guide/prototype-an-assistant.rst:81]:
```suggestion
    forms:
    - newsletter_form:
        email:
        - type: from_text
```
```suggestion
    forms:
    - newsletter_form:
        email:
        - type: from_text
```

File: docs/user-guide/prototype-an-assistant.rst [hunk 11]
```diff
@@ -0,0 +1,198 @@
...
+:ref:`Read about best practices for NLU <training-data-format>`
+
+
+3. Define a simple form
+~~~~~~~~~~~~~~~~~~~~~~~
+
+For most user goals, the bot will need to collect some information from the user to fulfill their request.
+To do so, we define a `form <glossary>`_. You can change the name of the form to reflect your user goal.
+Add to or replace the "email" item in the list below to reflect the information the bot needs
+to collect for your user goal. Leave the "type" field the same for any items you add.
```
Comment (erohmensing) [inline docs/user-guide/prototype-an-assistant.rst:70]:
```suggestion
Add to or replace the ``email`` item in the list below to reflect the information the bot needs
to collect for your user goal. Leave the ``type`` field the same for any items you add.
```

File: docs/user-guide/prototype-an-assistant.rst [hunk 12]
```diff
@@ -0,0 +1,198 @@
...
+    forms:
+    - newsletter_form:
+       email:
+       - type: from_text
+
+
+4. Write a rule
+~~~~~~~~~~~~~~~
+
+Rules are conversation snippets that should always precede in exactly the same way. The ellipsis in the
```
Comment (erohmensing) [inline docs/user-guide/prototype-an-assistant.rst:87]:
```suggestion
Rules are conversation snippets that should always proceed in exactly the same way. The ellipsis in the
```

File: docs/user-guide/prototype-an-assistant.rst [hunk 13]
```diff
@@ -0,0 +1,198 @@
...
+       - intent: subscribe
+       - action: newsletter_form
+       - form: newsletter_form
+
+
+5. Write some stories
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Stories are example conversations of how your assistant should handle a user’s intent in context.
+The first stories you write should follow the `happy path <glossary>` for your user goal.
```
Comment (erohmensing) [inline docs/user-guide/prototype-an-assistant.rst:110]:
```suggestion
The first stories you write should follow the :ref:`happy path <glossary>` for your user goal.
```

File: docs/user-guide/prototype-an-assistant.rst [hunk 14]
```diff
@@ -0,0 +1,198 @@
...
+
+A story contains one or more blocks of (user) intent and (bot) `actions <glossary>`_ or `responses <glossary>`_.
```
Comment (erohmensing) [inline docs/user-guide/prototype-an-assistant.rst:112]:
```suggestion
A story contains one or more blocks of (user) intent and (bot) :ref:`actions <glossary>` or :ref:`responses <glossary>`.
```

File: docs/user-guide/prototype-an-assistant.rst [hunk 15]
```diff
@@ -0,0 +1,198 @@
...
+    :id: responses
+    :height: 300
+    :tracking_endpoint: https://trainer-service.prototyping.rasa.com/startPrototyping
+
+    responses:
+       utter_greet:
+           - text: |
+               Hallo! How can I help you?
+           - text: |
+               hi!
```
Comment (erohmensing) [inline docs/user-guide/prototype-an-assistant.rst:157]:
```suggestion
               Hi!
```

File: docs/user-guide/prototype-an-assistant.rst [hunk 16]
```diff
@@ -0,0 +1,198 @@
...
+               I’ve subscribed {email} to the newsletter!
+           - text: |
+               You’ve been subscribed, the newsletter will be sent to {email}.
+
+.. note::
+
+    For this prototype, we have only defined responses, meaning the only thing the assistant does is respond with a
+    predefined message. Custom actions, however, can be defined to do whatever you’d like.
+    For example, for the user goal of subscribing to a newsletter, you could create a custom action
+    that adds the user’s email to a database. You can see an example of this in `Sara's action code <action code>`_.
```
Comment (erohmensing) [inline docs/user-guide/prototype-an-assistant.rst:172]: This needs a proper link

File: docs/user-guide/prototype-an-assistant.rst [hunk 17]
```diff
@@ -0,0 +1,198 @@
...
+7. Train and run
+~~~~~~~~~~~~~~~~
+
+Rasa has a command line interface that allows you to train and run your bot from a terminal.
+To train your bot on the NLU data, stories and responses you’ve just defined, run ``rasa train`` using
+the button below:
+
+.. train-button::
+    :endpoint: https://trainer-service.prototyping.rasa.com/trainings
+    :method: POST
```
Comment (erohmensing) [inline docs/user-guide/prototype-an-assistant.rst:184]: This takes a pretty long time -- where is the config stored? can we reduce the epochs?

File: docs/user-guide/prototype-an-assistant.rst [hunk 18]
```diff
@@ -0,0 +1,198 @@
...
+
+.. chat-block::
+
+
+8. What's next?
+~~~~~~~~~~~~~~~
+
+You can download this project and build on it to create a more advanced assistant.
+In your downloaded project, you’ll notice several files that were configured for you that you didn’t edit on this page.
+To learn more about configs, domains and actions, refer to the advanced tutorials.
```
Comment (erohmensing) [inline docs/user-guide/prototype-an-assistant.rst:196]: link to tutorials, or to their docs pages

File: docs/user-guide/prototype-an-assistant.rst (hunk 10 above)
Comment (erohmensing) [reply docs/user-guide/prototype-an-assistant.rst:81]: slash maybe in general we need to pick an indentation level and stick to it (there was 1, 2, and 4 i believe)

File: docs/conf.py (hunk 3 above)
Comment (tmbo) [reply docs/conf.py:44]: I've created a new ticket for @alwx to port these parts to the new docs :+1: https://github.com/RasaHQ/rasa/issues/6269

File: docs/index.rst (hunk 4 above)
Comment (amn41) [reply docs/index.rst:28]: yes

Comment (erohmensing) [reply docs/index.rst:28]:
```suggestion
   user-guide/prototype-an-assistant
   user-guide/building-assistants
```

File: docs/conf.py (hunk 3 above)
Comment (erohmensing) [reply docs/conf.py:44]: Awesome thanks 🙏

File: docs/user-guide/prototype-an-assistant.rst (hunk 17 above)
Comment (erohmensing) [reply docs/user-guide/prototype-an-assistant.rst:184]: Ah i guess we want to keep it the standard for when it's downloaded though

Comment (alwx) [reply docs/user-guide/prototype-an-assistant.rst:184]: I guess it's fine now

File: docs/user-guide/prototype-an-assistant.rst [hunk 19]
```diff
@@ -0,0 +1,200 @@
...
```
Comment (erohmensing) [inline docs/user-guide/prototype-an-assistant.rst:198]:
```suggestion
Check out other docs pages to learn more about :ref:`domains`, :ref:`actions`,
and your config's :ref:`pipeline <choosing-a-pipeline>` and :ref:`policies`.
```

**Pull request merged**